src/java/org/apache/cassandra/service/StorageProxy.java

```diff
@@ -795,6 +805,20 @@ protected Verb verb()
         });
     }
 
+    public static boolean hasLocalMutation(IMutation mutation)
+    {
+        String keyspace = mutation.getKeyspaceName();
+        List<String> endpoints = StorageService.instance.getNaturalEndpointsWithPort(keyspace, mutation.key().getKey());
+        return canDoLocalRequest(endpoints);
+    }
+
+    public static boolean canDoLocalRequest(List<String> endpoints)
+    {
+        String[] split = FBUtilities.getBroadcastAddressAndPort().toString().split("/");
```

Comment:
This is here because getBroadcastAddressAndPort returns something like "/127.0.0.1:1234" (with / at the start), there does not seem to be any other method which would return port as well without that slash. I would create a helper method where this would be precomputed and just reused.